### ISSUE TITLE
[Feature Request] add categorys

### ISSUE THREAD

**Issue by 6543** [helloworld1/FreeOTPPlus#85]
idear also came from https://github.com/jamie-mh/AuthenticatorPro ;)

**Comment by kdoteu** [comment]
At this moment I've 30 Entries for work, personal and hobby related stuff. It would be very nice to organize these entries, so its easier to find the right ones.

**Comment by NRGLine4Sec** [comment]
I agree.

**Comment by jugendhacker** [comment]
> At this moment I've 30 Entries for work, personal and hobby related stuff. It would be very nice to organize these entries, so its easier to find the right ones.

I mean there is the search.... but categories would be nice

**Comment by kdoteu** [comment]
> I mean there is the search.... but categories would be nice

Ohh never noticed the search. Sometimes I'm blind in my workflows. Thanks for the mentions. But for better organisation categories would be a nice feature.

**Comment by OmlineEditor** [comment]
I think you need to make folders in the likeness of folders and files on the computer. folders are like groups.

**Comment by Mx7ca** [comment]
I also agree that categories would be quite useful. I have many entries and it would be nice to have the possibility to separate them.

**Comment by whereismybugfix** [comment]
I'd like to add my five cents as I stumbled upon the same idea. I've got tons of entries that are difficult to manage without some kind of folder structure or categories. Thank you all for bringing this up in this GitHub issue thread! I wanted to open my own but found this one.

**Comment by mlippert** [comment]
I initially missed this issue, and started a discussion #252 . Categories would provide the same advantages I was thinking about.
I stated _it might even be better particularly if an entry could belong to multiple categories. Even if an entry can only belong to a single category, I think that is pretty much exactly what I was suggesting in this feature request._

I don't think there has been much discussion about specifics such as: can an entry belong to multiple categories or only 1.

**Comment by whereismybugfix** [comment]
> it might even be better particularly if an entry could belong to multiple categories

That's what I've been thinking of too. Would be extremely useful.

**Comment by keks24** [comment]
I second that!

Especially for two-factor authentication, I have no opportunity to seperate `private` from `work`; most of the time, companies are not willing to give out `company phones`...

**Comment by aog2000a** [comment]
Another "me too". 

By the way, i just noticed that it IS possible (in v3.1.1) to custom-sort the entries, nice! that already helps a lot. But some kind of organizing tokens in folders or setting tags to group them would be perfect. Thanks to the devels!

**Comment by Magicrafter13** [comment]
I would also appreciate categories/folders, mainly because organizing them right now isn't the easiest since if you drag to the bottom they just shoot all the way to the end, and then you have to slowly trudge them back up to where you want them.

**Comment by CPUGPU** [comment]
Expanding on this suggestion, the "groups" implementation in [Catima](https://github.com/CatimaLoyalty/Android) loyalty card manager (really a multi-purpose barcode manager) does a great job with groups/categories.  I especially like that it's easy to modify groups, and entries can be in multiple groups.  I don't know enough (any...) Java or Kotlin to say, but maybe some of that code can be reused?

**Comment by helloworld1** [comment]
Feature is added in https://github.com/helloworld1/FreeOTPPlus/commit/dd1c5ad8700ef80dad6dd8e13e5b78bf131128a1 and will be released in version 3.2

**Comment by keks24** [comment]
Awesome, thank you!
